c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -829,16 +837,25 @@ mfxStatus MfxC2DecoderComponent::InitDecoder(std::shared_ptr<C2BlockPool> c2_all
         }
 
         if (MFX_ERR_NONE == mfx_res) {
+            // set memory type according to consumer usage sent from framework
+            m_mfxVideoParams.IOPattern = (C2MemoryUsage::CPU_READ == m_consumerUsage) ?
```

Comment:
Whether the framework set the cosumer usage?  can you point out the code, I don't find any, but if so it is very good idea.

Reply:
Yes, framework will set the consumer usage https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/frameworks/av/media/codec2/sfplugin/CCodecBufferChannel.cpp#1348. This is different from Android R.

Reply:
Ok, Thanks!